### ISSUE TITLE
Feature Request: Support for Habana HPU Accelerators

### ISSUE THREAD

**Issue by htang2012** [ray-project/ray#37635]
### Description

Hello Ray maintainers and community,

we've been using Ray for my work and find it to be a valuable tool for scalable and distributed machine learning. I believe it would be beneficial to many users if support for Habana HPU accelerators were added.

Habana Labs, an Intel company, designs and manufactures specialized hardware accelerators for artificial intelligence workloads, known as Habana Processing Units (HPUs). Habana's accelerators are becoming more prevalent in data centers due to their high performance and efficiency. By adding support for these accelerators, Ray could tap into the growing community of researchers and engineers using Habana's hardware.

Support for HPUs could potentially be implemented in a similar way to how Ray currently supports other accelerators, such as CPUs, GPUs, and gloo, NCCL

Thank you for considering this feature request. I'm looking forward to any discussion on this matter.

Best regards,

Henry Tang

### Use case

Same as GPU and NCCL backend for distributed deployment. we would expect to run on HPU and HCCL (habana distributed backend).

**Comment by rkooo567** [comment]
https://github.com/ray-project/ray/pull/36493

we have an ongoing PR from intel under review to support XPU. Is this a different type of hardware?

**Comment by rkooo567** [comment]
https://github.com/ray-project/ray/blob/master/python/ray/util/accelerators/accelerators.md

-> it could be helpful if you are interested in contribution

**Comment by htang2012** [comment]
> #36493
> 
> we have an ongoing PR from intel under review to support XPU. Is this a different type of hardware?

Yes, Habana is a subdivision of Intel which was acquired in 2019 and its HPU and Intel GPU are different accelerators.
